[regression] isinstance complains about parametrized generics with any use of TypeAlias

**Bug Report**

Any attempt to call `isinstance` with a variable annotated as a `TypeAlias` seems to result in the error message `error: Parameterized generics cannot be used with class or instance checks`, even if the aliased type is non-generic. mypy_primer points at https://github.com/python/mypy/pull/18173 as the breaking commit, but that PR seems to have been intended to deal with type aliases of generic types; it seems that we're treating _all_ `TypeAlias`-based aliases as generic types, which AFAICT is broader than intended. 

**To Reproduce**

[MyPy playground link](https://mypy-play.net/?mypy=master&python=3.12&gist=dedafca242fde5949bad614b4668806f), using current master

```python
from typing import Any, TypeAlias

class Foo:
    pass

Alias: TypeAlias = Foo

def is_foo(x: Any) -> bool:
    return isinstance(x, Alias)
```

**Expected Behavior**

MyPy finds no errors, matching the runtime behavior. Pyright and mypy 1.4 both find no errors in my minimal example.

**Actual Behavior**

Per the playground link:

```
main.py:9: error: Parameterized generics cannot be used with class or instance checks  [misc]
Found 1 error in 1 file (checked 1 source file)
```

**Your Environment**

I found this when dogfooding mypy master commit c4e2eb79905d7b381db8caed161cebb04622ebc2 against Dropbox's internal Python repo, but mypy_primer also reproduces the same thing, as does mypy playground.

--- a/mypy/semanal.py
+++ b/mypy/semanal.py
@@ -4022,7 +4022,6 @@ class SemanticAnalyzer(
             and not res.args
             and not empty_tuple_index
             and not pep_695
-            and not pep_613
         )
         if isinstance(res, ProperType) and isinstance(res, Instance):
             if not validate_instance(res, self.fail, empty_tuple_index):

--- a/test-data/unit/check-type-aliases.test
+++ b/test-data/unit/check-type-aliases.test
@@ -1243,31 +1243,22 @@ A = Union[int, List[A]]
 def func(x: A) -> int: ...
 [builtins fixtures/tuple.pyi]
 
-[case testAliasExplicitNoArgsBasic]
-from typing import Any, List, assert_type
+[case testAliasNonGeneric]
 from typing_extensions import TypeAlias
+class Foo: ...
 
-Implicit = List
-Explicit: TypeAlias = List
+ImplicitFoo = Foo
+ExplicitFoo: TypeAlias = Foo
 
-x1: Implicit[str]
-x2: Explicit[str]  # E: Bad number of arguments for type alias, expected 0, given 1
-assert_type(x1, List[str])
-assert_type(x2, List[Any])
-[builtins fixtures/tuple.pyi]
+x1: ImplicitFoo[str]  # E: "Foo" expects no type arguments, but 1 given
+x2: ExplicitFoo[str]  # E: "Foo" expects no type arguments, but 1 given
 
-[case testAliasExplicitNoArgsGenericClass]
-# flags: --python-version 3.9
-from typing import Any, assert_type
-from typing_extensions import TypeAlias
+def is_foo(x: object):
+    if isinstance(x, ImplicitFoo):
+        pass
+    if isinstance(x, ExplicitFoo):
+        pass
 
-Implicit = list
-Explicit: TypeAlias = list
-
-x1: Implicit[str]
-x2: Explicit[str]  # E: Bad number of arguments for type alias, expected 0, given 1
-assert_type(x1, list[str])
-assert_type(x2, list[Any])
 [builtins fixtures/tuple.pyi]
 
 [case testAliasExplicitNoArgsTuple]

--- a/test-data/unit/diff.test
+++ b/test-data/unit/diff.test
@@ -1563,6 +1563,7 @@ type H[T] = int
 __main__.A
 __main__.C
 __main__.D
+__main__.E
 __main__.G
 __main__.H
 

